crypto/secp256r1/publickey.go

```diff
@@ -0,0 +1,26 @@
+package secp256r1
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"math/big"
+)
+
+// Generates appropriate public key format from given coordinates
+func newPublicKey(x, y *big.Int) *ecdsa.PublicKey {
+	// Check if the given coordinates are valid
+	if x == nil || y == nil || !elliptic.P256().IsOnCurve(x, y) {
+		return nil
+	}
+
+	// Check if the given coordinates are the reference point (infinity)
+	if x.Sign() == 0 && y.Sign() == 0 {
+		return nil
+	}
```

Comment:
@ulerdogan this check seems to be redundant as the library checks it [here](https://github.com/golang/go/blob/master/src/crypto/elliptic/nistec.go#L129-L134). I added a test to confirm this [here](https://github.com/ethereum-optimism/op-geth/pull/168/commits/34d76cb9717e78f8195325a253119e621ef57857#diff-3548292e7ee4a75fc8146397c6baf5c969f6fe6cd9355df322cdb4f11103e004R5463-R5467). 

Do you have any concerns about removing this?

Reply:
Thanks for improving, yes we can remove it! I will reflect the changes in my own branch.

Reply:
Perfect, I'll update this PR now. Thank you for all the help 🙏

Reply:
Thanks all!